### DISCUSSION TITLE
2fa verification problem

### DISCUSSION THREAD

**Discussion by ramimoghrabi** [sbertix/Swiftagram#264] [General]
i am doing the 2fa verification with the authenticator, but receiving error 400 see below some of the output :

invalidResponse(<NSHTTPURLResponse: 0x600001f070a0> { URL: https://i.instagram.com/api/v1/accounts/two_factor_login/ } { Status Code: 400, Headers {
    "Alt-Svc" =     (
        "h3=\":443\"; ma=3600,h3-29=\":443\"; ma=3600"
    );
    "Cache-Control" =     (
        "private, no-cache, no-store, must-revalidate"
    );
    "Content-Language" =     (
        en
    );
    "Content-Length" =     (
        123
    );
    "Content-Type" =     (
        "application/json; charset=utf-8"
    );
    Date =     (
        "Mon, 15 Nov 2021 21:29:09 GMT"
    );
    Expires =     (
        "Sat, 01 Jan 2000 00:00:00 GMT"
    );
    Pragma =     (
        "no-cache"
    );

What could be the problem ? thanks for the help in advance

**Comment by igornazarov1991** [comment]
I have another issue with 2fa: after calling 

`challenge.code(/* the code */).authenticate()`

 response seems to be good enough

> Optional<NSHTTPURLResponse>
  - some : <NSHTTPURLResponse: 0x600001fbef00> { URL: https://i.instagram.com/api/v1/accounts/two_factor_login/ } { Status Code: 200, Headers {
    ...
    
The problem is that the library produces the error

`Authenticator.Error.invalidResponse(result.response)`

when parsing this response. As far as I understand, it happens due to the following code in `Authenticator+TwoFactor` file:

`guard value.isEmpty, let response = result.response as? HTTPURLResponse else {
             throw Authenticator.Error.invalidResponse(result.response)
 }`
 
 From my perspective this code should check for `guard !value.isEmpty`, since in current implementation the logic is to report an error when the `value` is here, which is obviously an incorrect behaviour.
